libdnf/base/base.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/base/base.hpp"
+
+#include "libdnf/conf/config_parser.hpp"
+#include "libdnf/conf/const.hpp"
+
+#include <fmt/format.h>
+
+#include <algorithm>
+#include <filesystem>
+#include <vector>
+
+namespace libdnf {
+
+static void load_config_from_parser(
```

Comment:
Probably in future the function will require additional adgument - libdnf::Option::Priority. But it is a static function that could be changed any time.